extensions/amp-consent/0.1/tcf-api-commands.js

```diff
@@ -14,53 +14,147 @@
+      case TCF_POST_MESSAGE_API_COMMANDS.PING:
+        this.handlePingEvent_(payload, win);
+        break;
+      case TCF_POST_MESSAGE_API_COMMANDS.GET_TC_DATA:
+      case TCF_POST_MESSAGE_API_COMMANDS.ADD_EVENT_LISTENER:
+      case TCF_POST_MESSAGE_API_COMMANDS.REMOVE_EVENT_LISTENER:
+      default:
+        return;
+    }
   }
-}
 
-/**
- * Checks if the payload from the `tcfapiCall` is valid.
- * @param {JsonObject} payload
- * @return {boolean}
- */
-export function isValidTcfApiCall(payload) {
-  if (!isObject(payload)) {
-    user().error(TAG, `"tcfapiCall" is not an object: ${payload}`);
-    return false;
+  /**
+   * Create minimal PingReturn object. Send to original iframe
+   * once object has been filled.
+   * @param {!Object} payload
+   * @param {!Window} win
+   */
+  handlePingEvent_(payload, win) {
+    this.policyManager_.getConsentMetadataInfo('default').then((metadata) => {
+      const returnValue = this.getMinimalPingReturn_(metadata);
+      const {callId} = payload;
+
+      this.sendTcfApiReturn_(win, returnValue, callId);
+    });
   }
-  const {command, parameter, version} = payload;
-  if (!isEnumValue(TCF_POST_MESSAGE_API_COMMANDS, command)) {
-    user().error(TAG, `Unsupported command found in "tcfapiCall": ${command}`);
-    return false;
+
+  /**
+   * Create minimal PingReturn object.
+   * @param {?Object} metadata
+   * @return {!MinimalPingReturn}
+   */
+  getMinimalPingReturn_(metadata) {
+    const gdprApplies = metadata ? metadata['gdprApplies'] : undefined;
+    return {
+      gdprApplies,
+      cmpLoaded: CMP_LOADED,
+      cmpStatus: CMP_STATUS,
+      tcfPolicyVersion: TCF_POLICY_VERSION,
+    };
   }
-  if (parameter !== undefined) {
-    user().error(
-      TAG,
-      `Unsupported parameter found in "tcfapiCall": ${parameter}`
+
+  /**
+   *
+   * @param {!Window} win
+   * @param {!JsonObject} returnValue
+   * @param {string} callId
+   * @param {boolean=} success
+   */
+  sendTcfApiReturn_(win, returnValue, callId, success) {
+    if (!win) {
+      return;
+    }
+
+    const __tcfapiReturn = {returnValue, callId, success};
+    win./*OK*/ postMessage(
+      /** @type {!JsonObject} */ ({
+        __tcfapiReturn,
+      }),
+      '*'
     );
-    return false;
   }
-  if (version != '2') {
-    user().error(TAG, `Found incorrect version in "tcfapiCall": ${version}`);
-    return false;
+
+  /**
+   * Checks if the payload from the `tcfapiCall` is valid.
+   * @param {JsonObject} payload
+   * @return {boolean}
+   */
+  isValidTcfApiCall_(payload) {
+    if (!isObject(payload)) {
+      user().error(TAG, `"tcfapiCall" is not an object: ${payload}`);
+      return false;
+    }
+    const {command, parameter, version} = payload;
+    if (!isEnumValue(TCF_POST_MESSAGE_API_COMMANDS, command)) {
+      user().error(
+        TAG,
+        `Unsupported command found in "tcfapiCall": ${command}`
+      );
+      return false;
+    }
+    if (parameter !== undefined) {
```

Comment:
Is `null` an acceptable value for parameter? If no, then just change this to `if (parameter) {...}`